### PR TITLE
SecuredAuthorizationManager should cache annotation's value

### PR DESCRIPTION
**Pull request by evgeniycheban**
Closes gh-12232

### CODE REVIEW

File: core/src/test/java/org/springframework/security/authorization/AuthoritiesAuthorizationManagerTests.java [hunk 1]
```diff
@@ -74,14 +74,14 @@ void checkWhenUserHasNotAnyAuthorityThenDeniedDecision() {
 	}
 
 	@Test
-	void hasRoleWhenRoleHierarchySetThenGreaterRoleTakesPrecedence() {
+	void checkWhenRoleHierarchySetThenGreaterRoleTakesPrecedence() {
```
Comment (jzheaux) [inline core/src/test/java/org/springframework/security/authorization/AuthoritiesAuthorizationManagerTests.java:77]: Will you please put this in a separate commit with a comment that describes why the change was made? By doing that, each commit will stay focused on one atomic change.

**Review by jzheaux — CHANGES_REQUESTED** [review]
Thanks, @evgeniycheban! This looks great. I left a piece of feedback inline.

Also, now that we are in 2023, will you please update the copyright year for these files?

File: core/src/test/java/org/springframework/security/authorization/AuthoritiesAuthorizationManagerTests.java (hunk 1 above)
Comment (evgeniycheban) [reply core/src/test/java/org/springframework/security/authorization/AuthoritiesAuthorizationManagerTests.java:77]: Done.

**Comment by jzheaux** [comment]
Thanks, @evgeniycheban! This is now merged into `main`.
